modules/managers/galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.selenium.manager/src/main/java/dev/galasa/selenium/internal/FirefoxOptionsImpl.java

```diff
@@ -96,14 +104,19 @@ public void setCapability(String key, Object value) {
         options.setCapability(key, value);
     }
 
+    /**
+     * @deprecated Removed in Selenium 4. See https://www.selenium.dev/documentation/webdriver/troubleshooting/upgrade_to_selenium_4/
+     */
+    @Deprecated(forRemoval = true, since = "0.39.0")
     @Override
-    public void setLegacy(boolean bool) {
-        options.setLegacy(bool);
+    public void setLegacy(boolean isLegacy) {
+        options.setCapability("marionette", !isLegacy);
+        this.isLegacy = isLegacy;
```

Comment:
`setLegacy` no longer exists in Selenium 4 - see https://www.selenium.dev/documentation/webdriver/troubleshooting/upgrade_to_selenium_4/

Added a temporary implementation based on its previous 3.141.59 implementation in https://github.com/SeleniumHQ/selenium/blob/selenium-3.141.59/java/client/src/org/openqa/selenium/firefox/FirefoxOptions.java#L187-L190